packages/block-library/src/heading/edit.js

```diff
@@ -10,33 +15,37 @@ import { __ } from '@wordpress/i18n';
 import { Fragment } from '@wordpress/element';
 import { PanelBody } from '@wordpress/components';
 import { createBlock } from '@wordpress/blocks';
+import { withSelect } from '@wordpress/data';
 import {
 	RichText,
 	BlockControls,
 	InspectorControls,
 	AlignmentToolbar,
 } from '@wordpress/block-editor';
 
-export default function HeadingEdit( {
+function HeadingEdit( {
 	attributes,
 	setAttributes,
 	mergeBlocks,
 	insertBlocksAfter,
 	onReplace,
 	className,
+	levelChoices,
 } ) {
 	const { align, content, level, placeholder } = attributes;
 	const tagName = 'h' + level;
-
+	const BlockControlsLevelsRange = ( levelChoices.length <= 3 ) ?
+		levelChoices :
+		levelChoices.slice( 1, 4 ); //
```

Comment:
I'm not sure I'm following what the logic is here as far as the _second, third, fourth, and fifth_ level choice options being significant. I sense you'd considered to add an inline comment but forgot to complete it?

Taking a guess with the default options, is it mostly about limiting to a select few common options, and omitting `H1` specifically? In that case it seems more accurate to express as something like `take( without( levelChoices, [ 1 ] ), 3 )` (`Array#slice` can work just as well)

https://lodash.com/docs/4.17.11#take
https://lodash.com/docs/4.17.11#without

This also seems like it would help avoid having to test `length <= 3` and just have a single consistent expression for computing the level choices.

Reply:
@aduth What I'm trying to do here is to replicate the previous behavior in the BlockControls options (not the sidebar). Right now, it displays only 3 headings: H2, H3 and H4. So if the levels list contains 3 or less options, it will display all of them, otherwise, slice the array and display 3 elements starting by the second one. I think using Lodash for that would be an overkill for such a simple operation.